File: API.Docentes/API.Docentes.Domain/DTOs/Request/CreateDocenteRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace API.Docentes.Domain.DTOs.Request
{
    public class CreateDocenteRequest
    {
        public string? NombreCompleto { get; set; }
        public int? IdEscuela { get; set; }
        public int? IdFacultad { get; set; }
        public string? Email { get; set; }
        public string? NumeroDocumento { get; set; }
        public string? IdUsuario { get; set; }
    }
}

File: API.Presentacion/src/Docente/dto/create-docente.request.ts
import { ApiProperty } from "@nestjs/swagger";
import { IsNotEmpty, IsString, IsUUID, IsEmail } from "class-validator";


export class CreateDocenteRequest{


    @ApiProperty()
    @IsString()
    @IsNotEmpty()
    nombreCompleto:string

    @ApiProperty()
    @IsUUID()
    @IsNotEmpty()
    idEscuela:string

    @ApiProperty()
    @IsUUID()
    @IsNotEmpty()
    idFacultad:string

    @ApiProperty()
    @IsEmail()
    @IsNotEmpty()
    email:string;

    @ApiProperty()
    @IsString()
    @IsNotEmpty()
    numeroDocumento:string;
}